runtime/near-vm-logic/src/logic.rs

```diff
@@ -460,29 +386,36 @@ impl<'a> VMLogic<'a> {
     /// * If the registers exceed the memory limit returns `MemoryAccessViolation`.
     /// * If called as view function returns `ProhibitedInView`.
     pub fn predecessor_account_id(&mut self, register_id: u64) -> Result<()> {
-        let Self { context, registers, config, .. } = self;
+        let Self { context, registers, gas_counter, config, .. } = self;
+        gas_counter.pay_base(config.runtime_fees.ext_costs.predecessor_account_id)?;
         if context.is_view {
             return Err(HostError::ProhibitedInView("predecessor_account_id".to_string()).into());
         }
-        let data = context.predecessor_account_id.as_ref();
-        Self::internal_write_register(registers, config, register_id, data)
+        let data = context.predecessor_account_id.as_bytes();
+        gas_counter.pay_per_byte(
+            config.runtime_fees.ext_costs.predecessor_account_id_byte,
+            data.len() as u64,
+        )?;
+        Self::internal_write_register(registers, gas_counter, config, register_id, data)
     }
 
     /// Reads input to the contract call into the register. Input is expected to be in JSON-format.
     /// If input is provided saves the bytes (potentially zero) of input into register. If input is
     /// not provided makes the register "not used", i.e. `register_len` now returns `u64::MAX`.
     pub fn input(&mut self, register_id: u64) -> Result<()> {
-        let Self { context, registers, config, .. } = self;
-        Self::internal_write_register(registers, config, register_id, &context.input)
+        let Self { context, registers, gas_counter, config, .. } = self;
```

Comment:
We need to call `pay_base` here as we do with `predecessor_account_id`, for consistency.